src/helpers.cpp

```diff
@@ -151,4 +151,182 @@ Vec3 Triangulate2(const SE3 &g12, const Vec2 &xc1, const Vec2 &xc2) {
   return x;
 }
 
+
+Vec3 Triangulate3(const SE3 &g12, const Vec2 &xc1, const Vec2 &xc2) {
+
+  // Initalize the Rotation and Translation Matricies
+  Vec3 t12{g12.T()};
+  Mat3 R12{g12.R()};
+  Mat3 R21{R12.transpose()};
+  Vec3 t21{-1 * R12.transpose() * t12};
+
+  // Create homogeneous coordinates
+  Vec3 f0{xc1(0), xc1(1), 1.0};
+  f0.normalize();
```

Comment:
The `xc` inputs are already homogeneous coordinates. You shouldn't need to normalize again.

Reply:
I think it is a 2x1 vector. All the triangulate function definition has Vec2 as an input